apps/client/src/pages/time-table/hooks/use-timetable-festival-mutation.ts

```diff
@@ -20,16 +20,19 @@ export const useDeleteTimeTableFestival = () => {
   });
 };
 
-export const useAddTimeTableFestival = () => {
+export const useAddTimeTableFestival = (onSuccessCallback?: () => void) => {
   const queryClient = useQueryClient();
 
   return useMutation({
     mutationFn: (selectedFestivals: number[]) =>
       addFestivalTimeTable(selectedFestivals),
-    onSuccess: () => {
-      queryClient.invalidateQueries({
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
         queryKey: [...FESTIVAL_BUTTON_QUERY_KEY.ALL],
       });
```

Comment:
```suggestion
    onSuccess: () => {
      queryClient.removeQueries({
        queryKey: [...FESTIVAL_BUTTON_QUERY_KEY.ALL],
      });
```
UI에 타임테이블이 즉각적으로 반영되지 않는 이유는 비동기처리에 관한 문제가 아니라 쿼리키와 캐싱된 데이터가 제대로 초기화 되지 않기 때문이라고 생각해요.
불필요한 async-await 대신에 removeQueries를 사용하면 기존 쿼리키&캐시를 완전히 제거하고 최신화된 데이터와 UI를 유지할 수 있을 것 같아요.

Reply:
두 가지 코드 모두 UI가 즉시 반영되지 않던 문제는 해결되는 것 같아요 

하지만 `removeQueries` 를 사용해서 기존 쿼리 키와 캐시를 삭제하는 로직이 
festival 이 추가 될 때 마다 이루어지는 것이 더 적절한지는 모르겠어요

async await + `invalidateQueries` 로 해결이 가능하다면 이 방법이 더 적절하지 않을까요..? 어렵네요

Reply:
사실 클릭된 타임테이블은 state로 관리하고 있어서, 쿼리키와의 연관성은 잘 모르겠어요.. 두분이 제시한 방법으로 해결은 되는데 ui 관련 로직을 데이터 받아오는 훅에서 처리하는 게 맞나 싶기도 하고요
페스티벌 추가/삭제 시 데이터 갱신은 잘 되고 있는 걸로 봐서 상태 로직을 수정해줘야될 것 같긴 합니다 해당 state가 useButtonClicked 훅 내부에서 관리하고 있는데 임시로 useEffect 추가해봤더니 페스티벌 배열이 변경될 시 클릭된 타임테이블 state에 반영했더니 해결되는 것도 확인했습니다!! 

일단 지욱님 방법으로 당장의 오류는 해결하고 해당 로직 나중에 리팩토링하는 게 어떤지 싶네요
페스티벌이 변경될 때 클릭된 버튼 상태가 맞춰져야 하므로, 이 상태를 더 적절한 곳에서 관리하는 게 필요하다고 판단돼서 클릭된 타임테이블의 상태 관리 로직을 커스텀 훅이 아닌 컴포넌트 상위에서 컨트롤하는 구조가 나을거같아요